csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -824,7 +837,8 @@ public async Task SetPublishedEndpointsAsync(IEnumerable<Endpoint> newEndpoints)
                     }
                     else
                     {
-                        Protocol = fallbackProtocol;
+                        // This OA is mostly likely used for colocation, unless a router is set.
```

Comment:
If a router is set, wouldn't we use the router proxy's protocol as the adapter's protocol?